test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -0,0 +1,73 @@
+/*global JSON */
+define([
+  'jquery',
+  'vendor/lib/q',
+  'helpers/util/Expect',
+  'sandbox',
+  'util/DeviceFingerprint'
+],
+function ($, Q, Expect, $sandbox, DeviceFingerprint) {
+
+  Expect.describe('DeviceFingerprint', function () {
+
+    function mockIFrameMessages(success, errorMessage) {
+      var message = success ? {
+        type: 'FingerprintAvailable',
+        fingerprint: 'thisIsTheFingerprint'
+      } : errorMessage;
+      window.postMessage(JSON.stringify(message), '*');
+    }
+
+    beforeEach(function () {
+      $sandbox.append('<div class="auth-content"></div>');
+    });
+
+    it('iframe is created with the right src and it is hidden', function () {
+      spyOn(window, 'addEventListener');
+      DeviceFingerprint.generateDeviceFingerprint('baseUrl');
+      var $iFrame = $('#okta_fingerprint_iframe');
+      expect($iFrame).toExist();
+      expect($iFrame.attr('src')).toBe('baseUrl/auth/services/devicefingerprint');
+      expect($iFrame.is(':visible')).toBe(false);
+      expect(window.addEventListener).toHaveBeenCalledWith('message', jasmine.any(Function), false);
+    });
+
+    it('returns a fingerprint if the communication with the iframe is successfull', function (done) {
+      mockIFrameMessages(true);
+      DeviceFingerprint.generateDeviceFingerprint('file://')
+      .then(function (fingerprint) {
+        expect(fingerprint).not.toBeUndefined();
```

Comment:
not sure we need this line since the next line will check that it is not undefined (and explicitly, that it is that string)